test/audit_test.dart

```diff
@@ -31,73 +32,97 @@ void main() {
           errors = [];
           isDone = false;
           transformed = values.stream.audit(const Duration(milliseconds: 6));
+        });
+
+        void listen() {
           subscription = transformed
               .listen(emittedValues.add, onError: errors.add, onDone: () {
             isDone = true;
           });
-        });
+        }
 
         test('cancels values', () async {
+          listen();
           await subscription.cancel();
           expect(valuesCanceled, true);
         });
 
-        test('swallows values that come faster than duration', () async {
-          values
-            ..add(1)
-            ..add(2);
-          await values.close();
-          await waitForTimer(5);
-          expect(emittedValues, [2]);
+        test('swallows values that come faster than duration', () {
+          fakeAsync((async) {
+            listen();
```

Comment:
So nothing in `setUp` can be asynchronous (which is probably reasonable).

Maybe document that somewhere.

Reply:
Stuff in `setUp` can be async. I moved the call to `listen` because one of the new tests wants to do some work before listening.